### ISSUE TITLE
New Decoder/encoder available with pretty much all issues fixed, and huge speed increases.

### ISSUE THREAD

**Issue by GavinDarkglider** [jocover/jetson-ffmpeg#112]
Main tree:
https://gitlab.com/switchroot/switch-l4t-multimedia/FFmpeg

FFMPEG replacement deb for bionic, built against bionic ffmpeg sources to keep app compatibilty. Should use decoder by default in most apps avalable via apt. :
https://gitlab.com/l4t-community/gnu-linux/switchroot-pipeline/-/jobs/2211932828/artifacts/download?file_type=archive

VLC replacement deb(Needed a patch from upstream): https://gitlab.com/l4t-community/gnu-linux/switchroot-pipeline/-/jobs/2211932829/artifacts/download?file_type=archive

Replacement libffmpeg for chromium, sorry, no package yet: https://gitlab.azka.li/l4t-community/gnu-linux/switchroot-pipeline/-/jobs/647/artifacts/download?file_type=archive

For chromium, you should edit .desktop files, replace ```%U``` with  ```--no-sandbox --use-vulkan --enable-features=Vulkan %U```, then find/replace libffmpeg.so with one in archive.

Dont know off the top of my head where this is stored.

Also, Chrome doesnt do vp9/av1 via ffmpeg, so you need to install a plugin called h264ify if you want decoder to work in youtube, also this disables 4k. Sorry, Blame chrome developers for this.

**Comment by kevmo314** [comment]
I'm curious what's the source of the speed increase? I see that the new lib is using the V4L2 API instead of the multimedia API, is that it?

**Comment by eusoubrasileiro** [comment]
@GavinDarkglider broken links there. I was looking for your faster ffmpeg. Where is that?

**Comment by GavinDarkglider** [comment]
> @GavinDarkglider broken links there. I was looking for your faster ffmpeg. Where is that?

Yeah, I dont know what happened with the links, it was all gitlab ci stuff, the first link is the ffmpeg 4.4 branch, but you will need to back port it to the version from bionic if you want it to be system-wide. I will see what I can do about the links, but no promises.

**Comment by Azkali** [comment]
We have a repo setup for jetson: https://jetson.repo.azka.li
Everything cited above works on Jetson and is contained in the repo (repackaged chromium, vlc and ffmpeg) no need to edit the desktop file since the deb already embed those configuration in chromium's configs

**Comment by xsacha** [comment]
@Azkali is it possible to publish a new build on the repo with latest changes (especially the close stream memory leak)? Also, with rtsp enabled. Thanks.
Edit: I'll build my own anyway.

**Comment by gjrtimmer** [comment]
For anyone landing here looking for an actively maintained fork, [gjrtimmer/jetson-ffmpeg](https://github.com/gjrtimmer/jetson-ffmpeg) supports FFmpeg 6.0 through 8.1, both JetPack 5.x and 6.x, and includes decoder segfault guards, encoder concurrency fixes, frame pool lifecycle management, the new `receive_packet` encode API, DRM_PRIME zero-copy output, a VIC hardware scale filter, and MJPEG hardware decode/encode. All changes are regression-tested across the full FFmpeg version matrix on Jetson Orin hardware.
